Tests/UI/BillingForm/View/BillingFormTextFieldViewTests.swift

```diff
@@ -40,4 +40,41 @@ class BillingFormTextFieldViewTests: XCTestCase {
         XCTAssertEqual(view.textField.textColor, style.textfield.textColor)
         XCTAssertEqual(view.textField.tintColor, style.textfield.tintColor)
     }
+    
+    func testUpdateStyleFormatsPhoneNumber() {
+        var style = DefaultBillingFormPhoneNumberCellStyle()
+        style.textfield.text = "01206123123"
+        let view = BillingFormTextFieldView()
+        view.update(style: style, type: .phoneNumber(style), tag: 0)
+        
+        XCTAssertEqual(view.textField.text, "+44 1206 123123")
+    }
+    
+    func testShouldEndEditingIsFormattingDisplay() {
+        let testTextField = UITextField()
+        testTextField.text = "01206123123"
+        let view = BillingFormTextFieldView()
```

Comment:
is it possible to declare testTextField, view ( BillingFormTextFieldView() ) and style (DefaultBillingFormPhoneNumberCellStyle() ) externally to be reused in different test cases ?

Reply:
Each test case should build its own environment (as long as not overly complex - not the case here), I fundamentally disagree with the approach you're suggesting.